net/http/authz/authorizer.go

```diff
@@ -69,6 +74,35 @@ func accessTokenGuardData(grant *Grant) string {
 	return v.ID
 }
 
+func x509GuardData(grant *Grant) pkix.Name {
+	var v struct {
+		Subject struct {
+			CommonName         string   `json:"cn"`
+			OrganizationalUnit []string `json:"ou"`
```

Comment:
I think we should support the standard X.500 attributes (at least the MUSTs). From RFC 3280:

https://datatracker.ietf.org/doc/html/rfc3280#section-4.1.2.4

```

   Standard sets of attributes have been defined in the X.500 series of
   specifications [X.520].  Implementations of this specification MUST
   be prepared to receive the following standard attribute types in
   issuer and subject (section 4.1.2.6) names:

      * country,
      * organization,
      * organizational-unit,
      * distinguished name qualifier,
      * state or province name,
      * common name (e.g., "Susan Housley"), and
      * serial number.

   In addition, implementations of this specification SHOULD be prepared
   to receive the following standard attribute types in issuer and
   subject names:

      * locality,
      * title,
      * surname,
      * given name,
      * initials,
      * pseudonym, and
      * generation qualifier (e.g., "Jr.", "3rd", or "IV").
```